File: backend/gestion_commerciale/src/main/java/com/moonsystem/gestion_commerciale/services/ArticleService.java
package com.moonsystem.gestion_commerciale.services;

import java.math.BigDecimal;
import java.util.List;

import com.moonsystem.gestion_commerciale.dto.ArticleDto;
import com.moonsystem.gestion_commerciale.utils.PageResponse;

public interface ArticleService {

    ArticleDto save(ArticleDto dto);

    ArticleDto findByCod(Integer code);

    List<ArticleDto> findAll();

    ArticleDto update(Integer cod, ArticleDto dto);

    boolean delete(Integer code);

    List<String> findDistinctFamilles();

    List<String> findDistinctChoix();

    List<ArticleDto> search(String keyword);

    ArticleDto updateStock(Integer cod, BigDecimal newStock);

     PageResponse<ArticleDto> findAllPaginated(int page, int size);

     void checkUnicityDesignationAndChoix(ArticleDto dto,boolean Update);

     Integer getTotalElements();
     ArticleDto findByDesignAndChoix(String designation,String choix);
}

File: frontend/gestion-commerciale/src/app/services/article/article.service.ts
import { Injectable } from '@angular/core';
import {ArticleDto, ArticlesService, PageResponse} from "../../api-client";
import {Observable, of} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class ArticleService {

  constructor(private articleService: ArticlesService) { }
  getPaginatedArticles(page: number, size: number): Observable<PageResponse> {


    return this.articleService.getArticlesPaginated(page, size);
  }

  getTotalElements(){
    return this.articleService.getTotalElements();
  }


  supprimerArticle(cod:number){
    return this.articleService._delete(cod);
  }
  chercherArticleByKeyword(keyword:string){
    return this.articleService.searchArticles(keyword);
  }
  getArticleByCode(code:number){
    return this.articleService.findByCod(code);
  }
  creeArticle(article:ArticleDto):Observable<ArticleDto> {
    return this.articleService.save(article);
  }
  modifierArticle(code:number,article:ArticleDto):Observable<ArticleDto> {
    return this.articleService.update(code,article);
  }

  findAllMesArticles(){
    return this.articleService.findAll();
  }

  findAllFamilles():Observable<string[]>{
    return this.articleService.getFamilles();
  }
  findAllChoix():Observable<string[]>{
    return this.articleService.getChoix();
  }

  findArticleByDesignationAndChoix(designation:string,choix:string){
    return this.articleService.getArticleByDesAndChoix(designation,choix);
  }
  getAllDesognations(){
    return this.articleService.getAllDesignation();
  }

  getChoixByDes(des:string){
    return this.articleService.getChoixByDes(des);
  }
}
